src/fu-engine-request.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuEngine"
+
+#include "config.h"
+
+#include "fu-engine-request.h"
+
+struct _FuEngineRequest
+{
+	GObject			 parent_instance;
+	FwupdFeatureFlags	 feature_flags;
+	FwupdDeviceFlags	 device_flags;
+};
+
+G_DEFINE_TYPE (FuEngineRequest, fu_engine_request, G_TYPE_OBJECT)
+
+FwupdFeatureFlags
+fu_engine_request_get_feature_flags (FuEngineRequest *self)
+{
+	g_return_val_if_fail (FU_IS_ENGINE_REQUEST (self), FALSE);
+	return self->feature_flags;
+}
+
+void
+fu_engine_request_set_feature_flags (FuEngineRequest *self,
+				     FwupdFeatureFlags feature_flags)
+{
+	g_return_if_fail (FU_IS_ENGINE_REQUEST (self));
+	self->feature_flags = feature_flags;
+}
+
+FwupdDeviceFlags
+fu_engine_request_get_device_flags (FuEngineRequest *self)
+{
+	g_return_val_if_fail (FU_IS_ENGINE_REQUEST (self), FALSE);
+	return self->device_flags;
+}
+
+void
+fu_engine_request_set_device_flags (FuEngineRequest *self,
+				    FwupdDeviceFlags device_flags)
+{
+	g_return_if_fail (FU_IS_ENGINE_REQUEST (self));
+	self->device_flags = device_flags;
+}
+
+static void
+fu_engine_request_init (FuEngineRequest *self)
+{
+	/* untrusted */
+	self->device_flags = FWUPD_DEVICE_FLAG_NONE;
+
+	/* a subset of what fwupdmgr can do */
+	self->feature_flags = FWUPD_FEATURE_FLAG_DETACH_ACTION |
```

Comment:
I guess this is how it is specifically so that motd can show everything that `fwupdmgr` can do right?  I am a bit concerned that detail is going to get lost when a new feature comes up.  How about instead defaulting to `FWUPD_FEATURE_FLAG_NONE` and then setting these two in `fu-engine-helper.c`?

Reply:
Ya, that's probably better. Done.